aspnetcore/migration/webapi.md

```diff
@@ -1,126 +1,125 @@
 
-* Change any method returning *IHttpActionResult* to return a *IActionResult*
+    Attribute routing supports tokens, such as `[controller]` and `[action]`. At runtime, each token is replaced with the name of the controller or action, respectively, to which the attribute has been applied. The tokens reduce the number of magic strings in the project. The tokens also ensure routes remain synchronized with the corresponding controllers and actions when automatic rename refactorings are applied.
+1. Enable HTTP Get requests to the `ProductController` actions:
+    * Apply the [[HttpGet]](xref:Microsoft.AspNetCore.Mvc.HttpGetAttribute) attribute to the `GetAllProducts` action.
+    * Apply the `[HttpGet("{id}")]` attribute to the `GetProduct` action.
 
-Once these changes have been made and unused using statements removed, the migrated *ProductsController* class looks like this:
+After these changes and the removal of unused `using` statements, *ProductsController.cs* file looks like this:
 
-[!code-csharp[](../migration/webapi/sample/ProductsCore/Controllers/ProductsController.cs?highlight=1,2,6,8,9,27)]
+[!code-csharp[](webapi/sample/ProductsCore/Controllers/ProductsController.cs)]
 
-You should now be able to run the migrated project and browse to */api/products*; and, you should see the full list of 3 products. Browse to */api/products/1* and you should see the first product.
+Run the migrated project, and browse to `/api/products`. A full list of three products appears. Browse to `/api/products/1`. The first product appears.
 
-## ASP.NET 4.x Web API 2 compatibility shim
+## Compatibility shim
 
-A useful tool when migrating ASP.NET Web API projects to ASP.NET Core is the [Microsoft.AspNetCore.Mvc.WebApiCompatShim](https://www.nuget.org/packages/Microsoft.AspNetCore.Mvc.WebApiCompatShim) library. The compatibility shim extends ASP.NET Core to allow a number of different Web API 2 conventions to be used. The sample ported previously in this document is basic enough that the compatibility shim was not necessary. For larger projects, using the compatibility shim can be useful for temporarily bridging the API gap between ASP.NET Core and ASP.NET Web API 2.
+A useful tool when migrating ASP.NET 4.x Web API projects to ASP.NET Core is the [Microsoft.AspNetCore.Mvc.WebApiCompatShim](https://www.nuget.org/packages/Microsoft.AspNetCore.Mvc.WebApiCompatShim) library. The compatibility shim extends ASP.NET Core to support a number of conventions from ASP.NET 4.x Web API 2. The sample ported previously in this document is basic enough that the compatibility shim was unnecessary. For larger projects, using the compatibility shim can be useful for temporarily bridging the API gap between ASP.NET Core and ASP.NET 4.x Web API 2.
```

Comment:
IT's not a tool. how about something like:

The WebApiCompatShim provides a compatibility shim to move ASP.NET 4.x Web API projects to ASP.NET Core.